src/app/adminLayout/students/student-details/student-details.component.ts

```diff
@@ -15,24 +17,23 @@ export class StudentDetailsComponent implements OnInit {
   @Output() stopDisplayEvent = new EventEmitter<boolean>();
   @Output() newStudentEvent = new EventEmitter<Student>();
   @Output() deleteOldStudentEvent = new EventEmitter<number>();
-  stateOptions: any[];
+  stateOptions: any[] = [];
   editName: boolean;
   editLastName: boolean;
-  editEmail:boolean;
+  editEmail: boolean;
   updateForm: FormGroup;
+  inputTextElements: InputTextElement[] = [];
+  inputBooleanElements: InputBooleanElement[] = [];
 
   constructor(
     private readonly activatedRoute: ActivatedRoute,
     private readonly studentService: StudentService,
     private readonly formBuilder: FormBuilder,
   ) {
-    this.stateOptions = [
-      {label: 'tak', value: true},
-      {label: 'nie', value: false}
-    ];
-    this.editName=false;
-    this.editLastName=false;
-    this.editEmail=false;
+
+    this.editName = false;
+    this.editLastName = false;
```

Comment:
nie do końca mi się to chyba podoba, czy skoro updatujemy studenta i docelowo jest to wywołanie metody PUT? czyli nie musimy sprawdzać co jest updatowane tylko przesłać wszystko?

Reply:
a to już jest kwestia tylko i wyłącznie wyglądu, wymyśliłem sobie, że po przejściu do szczegółów kursanta pola będą edytowalne jedynie jak wciśniemy sobie ikonkę edycji i stąd te pola editCośtam, jak uważacie, że przekombinowane to wywalę. Może lepiej gdybym te pola nazwał editableName a nie editName, byłyby lepsze nazwy, no ale nazywanie zmiennych to najtrudniejsza część programowania jak twierdził jeden z naszych trenerów ;)

Reply:
spoko, można to zrobić tak że pola są po prostu zablokowane do edycji ale nie ustawiając tego pojedynczo dla każdego, narazie zostaw, coś do pojazdów zrobię to zobaczę, bo gdzieś widziałem blokowanie i odblokowanie wszystkiego na raz

Reply:
Może takie edytowanie wszystkiego naraz byłoby faktycznie lepsze, czyli mam jakąś listę -> przechodzę sobie w szczegóły -> mam nieedytowalny formularz -> klikam przycisk edytuj -> wszystkie pola mam edytowalne i pojawia mi się dopiero przycisk do zatwierdzenia zmian
Z tym blokowaniem edytowania to mój tok myślenia był taki, że pewnie zwykle te szczegóły kursanta służyłyby jednak do przeglądania a nie edycji, dlatego dobrze by się było zabezpieczyć przed jakimiś przypadkowymi zmianami